src/meta_schedule/task_scheduler/task_scheduler.cc

```diff
@@ -127,7 +128,20 @@ void TaskSchedulerNode::Tune() {
     ICHECK(!task->is_terminated);
     ICHECK(!task->runner_futures.defined());
     SearchStrategy strategy = task->search_strategy.value();
-    if ((task->measure_candidates = strategy->GenerateMeasureCandidates()).defined()) {
+    Optional<Array<MeasureCandidate>> candidates = strategy->GenerateMeasureCandidates();
+    if (candidates.defined()) {
+      // Filter out invalid candidates.
+      // An invalid candidate can arise due to a schedule application failure (e.g. tensorize).
+      Array<MeasureCandidate> valid_candidates;
+      for (MeasureCandidate candidate : candidates.value()) {
+        if (candidate.defined()) {
+          valid_candidates.push_back(candidate);
+        }
+      }
+
+      if (valid_candidates.size() == 0) continue;
```

Comment:
I'm not sure (1) if this case happens in practice, and (2) `continue` is the right thing to do here.

Reply:
Now that we consolidate the error handling logic in `ThreadedTraceApply`, do you think we don't need the logic here?

Reply:
Do you mean this line in particular or the whole change in this file?

Since the line below can return NullOpt, not all candidates from `GenerateMeasureCandidates()` are valid. So I think we still need the change in this file.
https://github.com/apache/tvm/blob/main/src/meta_schedule/search_strategy/replay_trace.cc#L138

Reply:
I can put that logic in `ReplayTraceNode::State::GenerateMeasureCandidates` instead, but then I also need to put the same logic in evo search and other space generators.

Reply:
On the system design, we assume that MeasureCandidates are all valid , otherwise it's not meaningful to send invalid ones to measure.

In terms of implementation: In `ReplayTrace`,  there is an infinite loop https://github.com/apache/tvm/blob/main/src/meta_schedule/search_strategy/replay_trace.cc#L134-L142 until it finds a valid schedule. The same applies to `EvolutionarySearch`.

Reply:
Yeah, I realized that after I put try/catch in `ThreadedApply`, all candidates are valid if ` support::parallel_for_dynamic` ever returns.

The issue I have now is that, that infinite loop can now really repeat forever, if some postprocs never succeed on all design spaces. I'm now trying conv2d VNNI auto-tensorize, and due to an implementation issue in auto-tensorize, `sch->Tensorize` in postproc always fail. But since the tensorize error is now caught by `ThreadedApply` and we repeat, it leads to an infinite loop.

So I'm now wondering if doing try/catch in `ThreadedApply` is the right solution here. Rather, we might want to wrap try/catch around `sch->Tensorize(...)` in a postproc rewrite, meaning we return non-tensorized schedule in case of auto-tensorization failure. I think this is what @vinx13 suggested last week.

So, if we agree that we should wrap try/catch around `sch->Tensorize(...)`, I don't need the change in this PR anymore. Do we have other good reason to put try/catch around `ThreadedApply`? Otherwise I'll close this.

Reply:
That sounds great to me Masa! Sorry somehow this PR got slipped through. I'm merging this immediately